docs/roles.md

```diff
@@ -0,0 +1,151 @@
+- Perform Transfer - `POST /transfers`
+- Retrieve Bulk Transfer Information - `GET /bulkTransfers/{ID}`
+- Perform Bulk Transfer - `POST /bulkTransfers`
+- Retrieve Transaction Information - `GET /transactions/{ID}`
+- Update Authorization Request - `PUT /authorizationRequests/{ID}` _[new pisp call]_
+
+**Inbound _(Note: some ommited for brevity)_**
+- Receive Authorization Creation Request - `POST /authorizationRequests/{ID}`
+
+
+### 2. PISP:
+**Outbound**
+- Lookup Party Information - `GET /parties/{Type}/{ID}, /parties/{Type}/{ID}/{SubId}`
+- Perform Authorization on this Thing `POST /authorizations/{ID}` [newly added for PISP case]
+- Update Authorization - `PUT /authorizations/{ID}`
+- Create Consent Request - `POST /consentRequests/{ID}`
+- Update Consent Request - `PUT /consentRequests/{ID}`
+- Update Consent - `POST /consents/{ID}`
+- Return Quote Information - `PUT /quotes/{ID}` _[this probably won't be needed since the quote is going to be a part of the authorizations callback]_
+- Return Quote Information Error - `PUT /quotes/{ID}/error` _[same as above]_
+- Get 3rd party request information - `GET /thirdPartyRequest/transaction/{ID}`
+- Initiate 3rd party request - `POST /thirdPartyRequest/transaction/{ID}`
+
+**Inbound**
+- Return Party Information - `PUT /parties/{Type}/{ID}, /parties/{Type}/{ID}/{SubId}`
+- Return Party Information Error - `PUT /parties/{Type}/{ID}/error, /parties/{Type}/{ID}/{SubId}/error`
+- Return Authorization Result - `PUT /authorizations/{ID}`
+- Return Authorization Error - `PUT /authorizations/{ID}/error`
+- Update Consent Request - `PUT /consentRequests/{ID`
+- Update Consent Request Error - `PUT /consentRequests/{ID}/error`
+- Update Consent - `POST /consents/{ID}`
+- Update 3rd Party Transaction Request - `PUT /thirdPartyRequest/transaction/{ID}`
+- Update 3rd Party Transaction Request Error - `PUT /thirdPartyRequest/transaction/{ID}/error`
+
+## Questions:
+
+- What endpoint are we using to get account balances/metadata? Should this be `GET /parties...` or something else, such as `thirdPartyRequests/balance`?
+- In the FSPIOP API Definition v1.1, `/authorizations` resource is used solely in conjunction with internal transaction requests. With the addition of a new PISP transaction requests `/thirdPartyRequests/transaction`, we may need to reconsider how `/authorizations` is defined so that it covers both the 'internal' (or existing) transaction request scenario, as well as the new 'Third Party' scenario
+
+## Performance Considerations:
+
+We assume Role Based Access Controls will be implemented at an API Gateway.
+
+In order to ensure such an implementation won't greatly affect performance, the goal is to have RBAC that need not inspect the _body_ of the API call (or at least minimize the number of API calls that depend on inspecting the body).
+
+This is possible with the PISP Case by using the a new resource for transaction requests: `thirdparty
```

Comment:
How and where in Hub we plan to implement new `thirdPartyRequests` resource? 
As a core component, or an additional service?
What about the ml-api-adapter? Will it be impacted?

Reply:
Any hint how will we implement `consents` and `consentRequests` resources? As one `consentsService`? Will this service be part of the Hub or some adapter?

Reply:
The plan is that the Open API definition will be amended to include the new thirdPartyRequests resource. The API adapter will, of course need to be adapted to support it, as is true for any change to the Open API specification.

I think that the plan will also be to add the consents and consentRequest resources to the Open API as well. Our thinking at the moment is that participants will send their requests to the hub and the hub will route them as required, including interacting with a central FIDO server where DFSPs do not implement their own FIDO server.

Reply:
> Any hint how will we implement `consents` and `consentRequests` resources? As one `consentsService`? Will this service be part of the Hub or some adapter?

I'm not sure what you mean by part of the Hub _or_ adapter, can you please clarify?

As for the implementation, my gut feeling (which needs further investigation) is that it will be something like a `ConsentsService` as you say, with perhaps a better name, provided that we can separate out its functions into a separate service appropriately.